packages/xdl/src/ProjectAssets.ts

```diff
@@ -349,3 +395,29 @@ async function collectAssets(
 
   return [...bundles.ios.assets, ...bundles.android.assets, ...manifestAssets];
 }
+
+export async function resolveAndCollectExpoUpdatesManifestAssets(
+  projectRoot: string,
+  exp: ExpoConfig,
+  urlResolver: (path: string) => string
+): Promise<{ url: string; hash: string; key: string; contentType: string }[]> {
+  const manifestAssets: { url: string; hash: string; key: string; contentType: string }[] = [];
+  await resolveExpoUpdatesManifestAssets({
+    projectRoot,
+    manifest: exp,
+    async assetKeyResolver(assetPath) {
+      const absolutePath = path.resolve(projectRoot, assetPath);
+      const contents = await fs.readFile(absolutePath);
+      const fileType = await FileType.fromBuffer(contents);
+      const hash = md5hex(contents);
+      manifestAssets.push({
+        url: urlResolver(assetPath),
+        hash,
+        key: assetPath,
+        contentType: fileType?.mime ?? '',
```

Comment:
Does EAS not accept null or omitted content types?

Reply:
In this case, EAS Update defaults to `application/octet-stream` (i.e. unknown bytes). I think it would be good for EAS Update and the development server to share this behavior so publishing doesn't change as many things.

Reply:
Ah, I see that we just use the extension in EAS cli and fall back to `application/octet-stream`: https://github.com/expo/eas-cli/blob/d50bdedcec70e8e001730ed6b4946d12b0c639ca/packages/eas-cli/src/project/publish.ts

I'll use that strategy.